### ISSUE TITLE
Oauth2.error not use received status_code

### ISSUE THREAD

**Issue by ermakoves** [OpenIDC/pyoidc#466]
https://github.com/OpenIDC/pyoidc/blob/9693b3c3556980ecd6b91639701c45f494998b0d/src/oic/oauth2/__init__.py#L96-L98

I think should be:
```
def error(error, descr=None, status_code=None):
    stat_txt = R2C.get(status, R2C[400])._status
    return error_response(error=error, descr=descr, status=stat_txt)
```

**Comment by tpazderka** [comment]
Thanks for your report!

Yes, that is correct, I think we should decide for either `error` or `error_response` and deprecate the other as well.

**Comment by lwm** [comment]
Happily accepting PRs!

**Comment by tpazderka** [comment]
Btw there is an inconsistency between the two methods. One includes the numeric code in the status and the other one does not.

I think we should consider changing the `Response` class to include only the integer representation of the status. The translation to text can be done in client side.

**Comment by tpazderka** [comment]
Solved in #491.
